client/a8c-for-agencies/sections/partner-directory/agency-details/index.tsx

```diff
@@ -53,15 +56,32 @@ const AgencyDetailsForm = ( { initialFormData }: Props ) => {
 				duration: 6000,
 			} )
 		);
-	}, [ page, reduxDispatch, translate ] );
+	}, [ translate ] );
 
-	const { formData, setFormData, isValidFormData } = useDetailsForm( {
+	const { formData, setFormData } = useDetailsForm( {
 		initialFormData,
 	} );
 	const { countryOptions } = useCountryList();
 
 	const { onSubmit, isSubmitting } = useSubmitForm( { formData, onSubmitSuccess, onSubmitError } );
 
+	const submitForm = () => {
+		const error = validate( formData );
+		if ( error ) {
+			//FIXME: check if there's a better way to distinct parent for scrolling to the top
+			const parent = document.getElementsByClassName( 'partner-directory__body' )?.[ 0 ];
+			// Scrolling only for fields positioned on top
+			if ( error.name || error.email || error.website || error.bio || error.logo ) {
+				if ( parent ) {
+					parent?.scrollTo( { behavior: 'smooth', top: 0 } );
+				}
+			}
+			return;
+		}
+
+		onSubmit();
+	};
+
```

Comment:
Maybe we could adopt Filipe's suggestion of validating when the input loses the focus (onBlur).

Reply:
Also, the agency details could be saved, even if they don't have all the fields filled out or are invalid.